AlgebraInLean/Basic.lean

```diff
@@ -1 +1,50 @@
-def hello := "world"
\ No newline at end of file
+import Mathlib.Tactic
+
+namespace Defs
+  class Magma (G : Type*) where
+    op : G → G → G
+
+  -- FIXME: is this actually helpful/necessary
+  attribute [always_inline] Magma.op
+
+  infixl:75 " ⋆ " => Magma.op
+
+  class Semigroup (G : Type*) extends Magma G where
+    protected assoc : ∀ a b c : G, (a ⋆ b) ⋆ c = a ⋆ (b ⋆ c)
+
+  @[simp]
+  theorem assoc [Semigroup G] (a b c : G) : (a ⋆ b) ⋆ c = a ⋆ (b ⋆ c) := Semigroup.assoc a b c
+
+  class Monoid (G : Type*) extends Semigroup G where
+    protected id : G
+    protected mul_id : ∀ a : G, a ⋆ id = a
+    protected id_mul : ∀ a : G, id ⋆ a = a
+
+  notation:max "𝕖" => Monoid.id
+
+  @[simp]
+  theorem mul_id [Monoid M] (a : M) : a ⋆ 𝕖 = a := Monoid.mul_id a
```

Comment:
this should be op_id

Reply:
I agree, and similarly for the one below (`id_mul`).